build.rs

```diff
@@ -42,44 +41,103 @@ fn main() {
+        let entries = match fs::read_dir(dir) {
+            Ok(e) => e,
+            Err(_) => continue };
+        for entry in entries {
+            let entry = entry?;
+            let ty = entry.file_type()?;
+            if ty.is_file() {
+                if predicate(&entry.file_name()) {
+                    return Ok(true)
+                }
+            } else if ty.is_dir() {
+                dirs.push(entry.path())
+            }
+            // Ignore symlinks
+        }
+    }
+    Ok(false)
+}
+
+
+fn main() {
+    // First, we build the SUNDIALS library, with requested modules with CMake
+
+    let mut lib_loc;
+    let mut inc_dir;
+    let mut library_type = "dylib";
     if cfg!(feature = "build_libraries") {
-        let dst = cmake::Config::new("vendor")
-            .define("CMAKE_INSTALL_LIBDIR", "lib")
-            .define("BUILD_STATIC_LIBS", static_libraries)
-            .define("BUILD_SHARED_LIBS", shared_libraries)
-            .define("BUILD_TESTING", "OFF")
-            .define("EXAMPLES_INSTALL", "OFF")
-            .define("EXAMPLES_ENABLE_C", "OFF")
-            .define("BUILD_ARKODE", feature!("arkode"))
-            .define("BUILD_CVODE", feature!("cvode"))
-            .define("BUILD_CVODES", feature!("cvodes"))
-            .define("BUILD_IDA", feature!("ida"))
-            .define("BUILD_IDAS", feature!("idas"))
-            .define("BUILD_KINSOL", feature!("kinsol"))
-            .define("OPENMP_ENABLE", feature!("nvecopenmp"))
-            .define("PTHREAD_ENABLE", feature!("nvecpthreads"))
-            .build();
-        let dst_disp = dst.display();
-        lib_loc = Some(format!("{}/lib", dst_disp));
-        inc_dir = Some(format!("{}/include", dst_disp));
+        (lib_loc, inc_dir, library_type) = build_vendored_sundials();
     } else {
         lib_loc = get_env_var("SUNDIALS_LIBRARY_DIR");
         inc_dir = get_env_var("SUNDIALS_INCLUDE_DIR");
     }
 
+    if lib_loc.is_none() && inc_dir.is_none() {
+        // No path specified, try to detect if the library is present
+        // on the system.
+        #[cfg(target_family = "unix")] {
+            // Try to find Sundials header files.
+            let std_inc = ["/usr/local/include/", "/usr/include/",
+                           "/usr/include/x86_64-linux-gnu",
+                           "/usr/local/Cellar"];
+            let predicate = |s: &OsString| {
+                s == "nvector_serial.h" };
+            let found = std_inc.iter().any(|d| {
+                is_in_dir(d, predicate).unwrap() });
+            if ! found {
+                (lib_loc, inc_dir, library_type) = build_vendored_sundials();
+            }
+        }
+        #[cfg(target_family = "windows")] {
+            let vcpkg = vcpkg::Config::new()
+                .emit_includes(true)
+                .find_package("sundials");
+            if vcpkg.is_err() {
+                (lib_loc, inc_dir, library_type) = build_vendored_sundials();
+            }
+        }
```

Comment:
I know nothing about windows so I'll trust you on this one.

Reply:
As I said, I do not work on Windows but this seems the way to go IMHO (and the [compilation & tests work](https://github.com/Chris00/sundials-sys/actions/runs/3768269115/jobs/6406563622)).